deploy/yamls/lvmsnapshot-crd.yaml

```
@@ -53,9 +53,17 @@ spec:
                   can not be edited after the snapshot has been provisioned.
                 minLength: 1
                 type: string
+              snapOriginVolumeCapacity:
+                description: SnapOriginVolumeCapacity represents the capacity of the
+                  origin volume at the time the snapshot was created.
+                type: string
```

Can we modify code to generate this?

```suggestion
                pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                type: string
                x-kubernetes-int-or-string: true
```

---

Btw do we need this on the spec?
Also is there a race in case volume gets resized before snapshot is created?

---

Shall we have it as int64?

---

We can have it as [Quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/), like [here](https://github.com/openebs/lvm-localpv/blob/f103ba5e3fa2521a3bc3e6e9c53608942b950b5d/pkg/apis/openebs.io/lvm/v1alpha1/lvmnode.go#L55).

---

Updated `snapOriginVolumeCapacity ` type to [Quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/)